[PENDING APPROVAL] Big Science map changes, misc map fixes, bugfixes and more.

ThePainkiller opened the pull request:
## Progress, at any cost. Including the contempt of your fellow colonist.

As this PR is quite big in terms of map changes, I will have to post **a lot** of  screenshots to provide a visual guide throughout all of them, with a short summary of what's changing. Click on **Expand** to see all visual guides individually without cluttering your scrolling experience, and verify with the map diff if necessary.

### NEW: Small library east of Hydroponics

<details> <summary> Expand </summary>

![Library_Botany](https://user-images.githubusercontent.com/1294508/175765403-767492ad-bbc9-4f4b-8b59-039e54cb3d5b.png)

> Adds a small library to the east of hydroponics, complete with power, atmos and fire safety measures. Adds an extinguisher to botany, as well as a red indicator under the table to make access to botanist room more obvious.
</details>

### TWEAK: Medical Lobby 

<details> <summary> Expand </summary>

![Medbay_Entrance](https://user-images.githubusercontent.com/1294508/175765620-0a32e857-f82d-4523-944f-3b2c0999c596.png)

> RND is moved further north. Medical Lobby is now almost exclusively medical side, with a north entrance for Science as well. Moved around the ATM closer to Chemistry Desk, removes one MiniPharma in lieu of a Disposals Bin for those colonists that always litter implanters after buying death implants. Adds two vending machines to the hallway, and proper firelocks for all windows and doors, as well as a fire extinguisher.

</details>

### Where did RND go, then?

<details> <summary> Expand </summary>

![RND_Graves_Breakroom](https://user-images.githubusercontent.com/1294508/175765790-b2fe57c8-6462-45fc-b2b6-74a20abc2ed4.png)

> Research & Development is now further north, as well as Tea and Graves' pond. The Break Room for both Scientists and Doctors alike rests where RND used to be, complete with a designated smoking corner and connection to both sides of the sister departments and Borg Storage. Ample of space and tables to accommodate stuff, and display racks to show anything for sale to the direct public. An ATM is nearby for ease of payment, and Robotics rests further north towards the Plasma Tag arena now. Everything OSHAcat compliant where fire safety measures are involved.

</details>

### On to Robotics next. What became of it?

<details> <summary> Expand </summary>

![Robotics_Plasmatag](https://user-images.githubusercontent.com/1294508/175765970-77e07a1c-04a5-46f8-abca-eb619fa83b46.png)

>Robotics got pushed further north, and given ample space to accommodate for the building of at least four simultaneous Mechs in its bay, complete with the usual recharging bay, a nearby ATM, and their own desk for their own particular transactions. It's got everything it always had, with proper polarized glass for their surgery rooms and all fire safeties taken care of. It is connected to RND to the south, and the new Testing Range to the east. Oil stains are varedited to be purely cosmetic, without leaving footprints behind, on request of r4d6.

</details>

### ...You said something about a Testing Range?

<details> <summary> Expand </summary>

![Testrange_robo_xflora](https://user-images.githubusercontent.com/1294508/175766139-07f4320e-4b04-40e1-be9e-ee400f65e0ee.png)

> Yes! A whole new area north of Xenoflora and east of Robotics, it is a dedicated range with reinforced walls and impassable rock at the end, designed for the testing of Soteria weaponry, both portable and Mech alike. Fully powered, independent, OSHA compliant for fire safety, accessible from the south for anyone with xenobio/flora access, and from the west with Robotics access, this features the elimination of that long hallway in lieu of a maintenance hallway for SMES wiring, as well as mechanical shutters to prevent roaches seeing you through glass, breaking it and ruining your day. Three chairs rest to the south to accommodate for any potential buyers to watch how well Soteria weaponry responds to their needs.

![xenoflora](https://user-images.githubusercontent.com/1294508/175766741-5457ae4d-f2c7-4245-9610-fd12c12cd9d1.png)

> Xenoflora, as stated before, got better anti-roach and anti-fire measures (firelocks) where they lacked them before. Remember, only YOU can prevent plasmafires from badly contained mutated plants!

</details>

### So, the long hallway is gone, for wiring?

<details> <summary> Expand </summary>

![RoboStorage_Hallway_Viro_Hydrogen](https://user-images.githubusercontent.com/1294508/175766389-6f527522-c6d7-47f0-882f-1bf46d145882.png)

> Yes, correct! Now it's a maintenance extension cord for powering Soteria, with a proper warning and a shutter, to also prevent more roaches from ruining your department (learned this through localtesting...), with its corresponding button directly south of it, to allow Guild members through for any potentially needed repairs in that area. It also encompasses now the Borg Storage wing, and the storage for the Hydrogen generator is now properly wired, APC connected, and part of the Research Area. Virology got its doors renamed and switched around its access to check for virology and xenobiology, total access unchanged to it. But there's more below...

</details>

### Continuing on Virology...

<details> <summary> Expand </summary>

![viro_lift](https://user-images.githubusercontent.com/1294508/175766524-c27bab1d-6692-4cc7-aa00-c1d2d44c4f36.png)

> It now has all proper firelocks, and part of its windows is now part of the lift, for more cohesion on fire response and firelocks being closed, as stated by the red outline representing the "Science lift area". Medical equipment now has properly labelled airlocks with actual access locked to medical personnel as well, and surgery gains two neat signs to indicate it's a place of surgery, if it wasn't obvious for the tools and surgery table already.

</details>

### What about Xenobio? Didn't you do touches up on it before?

<details> <summary> Expand </summary>

![Xenobio](https://user-images.githubusercontent.com/1294508/175766599-1125b0eb-7ea9-41bf-a3a2-0188d398ac73.png)

> Yes, but it could always be improved, at least safety wise! Now it has properly placed firelocks, and TWO fire alarms in case you get trapped from either side. Of course, both sides have extinguishers, though the slime pens side gets favored with a full locker to the easternmost end of it.

</details>

### Hold on, measures don't add up, what happened to the basketball court?

<details> <summary> Expand </summary>

![Basket_court](https://user-images.githubusercontent.com/1294508/175766848-ddc657ad-68f5-40cc-9370-523513d62cd2.png)

> Relax. It got shifted 1 tile south and west to accommodate for the library, and allow better uninterrupted flow straight into the Medical lobby, and also got a floor light to fix its poor lightning. Nothing suspicious going on here, nothing at all. . .

</details>

### Wait, not only the main Omni map was touched, what happened to the Science Outpost?

<details> <summary> You bet every part gets the OSHA glow up </summary>

![outpost_dorms_canisters](https://user-images.githubusercontent.com/1294508/175767684-259c2663-3a23-432e-b10c-8c7f52e59d76.png)

> Outpost was HEAVILY lacking a way to siphon its long, LONG hallways, so it desperately needed some scrubbers and filters for refilling, consindering all of the dangerously volatile gasses that get stored there. Speaking of which, proper signals have been added to better accentuate the danger these canisters pose to life itself.
The break room also gets a SweatMax removed (we know nerds do not lift) in lieu of a proper firelock closure, since the relaxation area is its own little area as well. Firelocks on each door of the dorms too.

![outpost_fire_locker](https://user-images.githubusercontent.com/1294508/175767789-e3e608f1-6bdb-4cd2-80cf-2139dfddbd90.png)

> And due to the danger of gasses, a wee extinguisher isn't enough. The toxins lab deserves a good 'ol locker with a firesuit to help the poor scientist that did an oopsie and flooded the whole room with gases the temperature of the Amethian sun. This also moves aside the light tube to better accomodate the locker.

![outpost_bombrange](https://user-images.githubusercontent.com/1294508/175767865-fa1fb2da-7eeb-4cbe-839c-017230ef8c96.png)

> For the aspiring Soteria demolitionist following Jeane's steps that's new to the colony, a signal to better indicate the place to test their hopefully safely contained explosive tanks.

![outpost_lift](https://user-images.githubusercontent.com/1294508/175767924-4d75e1ab-be9d-4d91-8d6e-85c98478bb60.png)

> The lift gets an OSHA QOL change of getting firelocks right off the bat near the lift, so that rescuing parties of guildies can properly gear up and set up inflatable barriers to better stop gases from spreading further. The fire alarm gets moved to this south corner for that particular reason.

![outpost_medical](https://user-images.githubusercontent.com/1294508/175767959-66d40881-8a97-4c83-8c81-a1d7881bf5b5.png)

> The long-abandoned medical corner of the outpost deserves to be undusted, getting firelocks to function as its own safe haven during a crisis, better rotated Sleepers (that face the right direction this time), and a proper large cell to power up that forgotten defibrillator someone tossed down ages ago.

![outpost_floodlight](https://user-images.githubusercontent.com/1294508/175768020-a1c0910f-6d45-4b60-bc0b-d5ad4c35ac16.png)

> And to finalize, something requested by Aeger as a small quality of life: A Floodlight for Xenoarcheology.

</details>


### I've had enough of Soteria! What else you claim you have "fixed" around this time?

<details> <summary> Relax. Most of the chaos is over. Most... </summary>

![Armory_range](https://user-images.githubusercontent.com/1294508/175766966-57718bae-fde5-4355-958c-97f8b1dc8ef9.png)

> Marshal range got its impenetrable rock back to prevent ricochet and overpenetrating a random colonist going through maintenance at a very unfortunate time. Also fixes floating cameras and lights atop airlocks, accommodating them around as fit. Also, more firelocks to prevent plasmafires.

![Marshallentrance](https://user-images.githubusercontent.com/1294508/175767119-8a3836e1-a09f-4a54-9e1a-7333203d482a.png)

> Marshal Brig entrance also had a bad case of floating light tubes, so floor lights are an acceptable solution.

![Plasmatag](https://user-images.githubusercontent.com/1294508/175767155-3ae74d45-31f6-43c3-aa48-e3a95480821d.png)

> Oh and the plasmatag arena got the floor light fix as well. Thought there are more changes I'd do to its asymmetric layout if it were up to me...

![floating_cam](https://user-images.githubusercontent.com/1294508/175767225-b27b59ae-b5e1-4ca2-b1e1-c691e774787f.png)

> Fixed a floating cam as well near the Premier's desk.

![Bar](https://user-images.githubusercontent.com/1294508/175767249-2398b84e-275f-402a-bc36-161f0a282f83.png)

> And speaking of non-map fixes, the Bar now has a cocktail showcase on its north eastern corner, which now CORRECTLY functions as a custom vendor for bartenders to mix and set drinks for sale during rush hours. Bar also gets the OSHA treatment of being smothered on firelocks to safely contain any fires that start within it (hopefully without the bartender inside...)

![Barviparea](https://user-images.githubusercontent.com/1294508/175767480-66806d8d-c14d-46da-9cfb-9e1cad01925c.png)

> Continuing with the bar, the VIP area upstairs had its maintenance doors properly powered by virtue of becoming part of its area. Not pictured in this outdated screenshot are two grilles with barricades behind the doors, and a Tau cross on the wall to the east for pest control, to prevent roaches from coming inside the luxurious area.

![maintAPC](https://user-images.githubusercontent.com/1294508/175767553-159280ea-00c0-41e6-9ad9-237e72ec228d.png)

> And finally, pointed out by Leudoberct on the discord, this (optional) APC in maintenance that powers the area was disconnected from the grid, now it has proper wiring across the loooong platform due south, connecting with the SMES near the Abandoned Garden.

</details>

### EDIT: More OSHA compliant bar backrooms

<details> <summary> Expand </summary>

![image](https://user-images.githubusercontent.com/1294508/175792846-d2b26c84-18dd-4b06-9fd1-aa9a3c4ba5d2.png)

> Replaced O2 locker with a proper firefighting locker, added a fire alarm based on the bar backroom area, adds firelocks to the windows leading to maintenance, and shutters to prevent the termite spawn from breaking in. Adds an extinguisher to the Bar backroom as well, since the only firefighting cabinet is on the bathroom.

</details>

### EDIT: Gardener gets a custom vendomat as requested by Wilson

<details> <summary> Expand </summary>

![image](https://user-images.githubusercontent.com/1294508/175800188-1575b36b-a4fa-48fb-aa48-fd6f9aa6bc5c.png)

> On the top floor of Hydroponics, in a table. All the relevant parts to build one are there as well.

</details>

**And that's it for map changes!**

<hr>

### Not-screenshotted map changes:

- Genetics lab got properly gated access gates locked behind Main Soteria Access, renamed as "Xenogenetics Lab" for the distinction of the old "Genetics Lab" (now named thus) in its northeastern corner
- CBO office and Psych office given the Osha treatment with actual firelocks for their windows
- Bar, Kitchen, and Bar's VIP rooms now have shared access between Bartender and Chef. This also means that Security personnel now has proper access to these previously locked "private" rooms for the sake of warrants, as they're not protected by privacy laws like Dormitories proper. 
- If there was more, I forgor.


<hr>

### On the non-mapping side of this PR, it does the following:

- Drink Showcase and Cocktail Showcase no longer dense
- Fixes a bug in which anyone could unlock and alter a custom vendor's access just by swiping their ID despite not having proper access to it
- Fixes Cocktal Showcase not properly overlaying its filled icon state

<hr>

## THIS PR IS PENDING LICH APPROVAL. IT WILL BE EDITED WITH PROOF OF APPROVAL ONCE AND IF I GET IT. DO NOT MERGE UNTIL THEN.

<hr>

## Changelog

:cl:
add: Added small Library and Soteria Testing Range to the map.
add: Fully functional Cocktail Showcase to the bar!
add: Library blurbs!
tweak: MASSIVE RND map changes, check the PR for full description and screenshots!
fix: The colony is now more OSHA compliant than ever! Make the science catto proud!
fix: Fixes floating cameras and lights on windows, and bad maintenance wiring.
fix: Fixes a fatal bug on all custom vendors that allowed anyone to swipe their ID to change the ownership of it.
soundadd: Added a couple short tracks as area ambience for the Library
server: Makes everyone hate me for this unwarranted change not most will like.
/:cl:

Spoopy11 commented:
Not a fan of the layout.

This alone is bad:
![image](https://user-images.githubusercontent.com/10091683/175769182-b77f0c7a-3888-4dd9-b178-4075254196e4.png)
You shouldn't go through a breakroom to reach a main lab.

![image](https://user-images.githubusercontent.com/10091683/175769236-b778cfbe-5d1c-4e9e-83a8-f1ae852ec7e1.png)
Why is there so much space being used when Robotics is already squat and pleasant  enough?

Having the Science and medical lobbies in one place promotes a certain unity between the departments, by separating them you force them apart and cause less traffic for them both; traffic that could generate roleplay opportunities even in the form of chatter.

Just my two quick cents.

ThePainkiller commented:
> You shouldn't go through a breakroom to reach a main lab.

Are you...ignoring the main door on the north?

![image](https://user-images.githubusercontent.com/1294508/175769468-ff2c2fa1-1854-489a-a949-83ee0e9925e4.png)

TheSleepyCatmom commented:
The R&D lab really didn't need to be modified like this, and it sorta just makes it look very strange. I agree that it should be connected to the lobby, same as medical, instead of just a desk pointed to the main hallway. They've always generally had a lobby, and even if we go by the old map's standards they were directly across from one another from a 3 tile hallway and not.. well, completely down the way like that. If you want to separate where they connect to the lobby or something, you should just make their own lobby and not keep both areas connected like that. Just looks... weird. Not ergonomic, not great to go through.

The rest of it looks fine imo.

Dimasw99 commented:
Having RnD and medical lobby combined encouraged a lot of RP and interactions between the departments. Chemistry could look over to RnD desk to chit chat or with patients and so on. Having them this separated is just going to hurt all those naturally developing interactions. 

Likewise, this changes practically nothing about the RnD area except that you now have moved a things around and gave it a bigger room that by itself is just restrictive again with that giant table in the middle creating a 2 tile wide gap around the RnD computer. Old RnD had a 3 tile wide area. 

This is no improvement and throws in some pretty bad RP down sides making it feel very sterile. 

Also I am not sure if you are trolling here but a neko plush. Really?

Dopiotl commented:
The others pretty much summed up what I'd say.

I'd be just rehearsing what has already been said against in a different flavor.

Please, Stop.

cdb-is-not-good commented:
Did...you use active scrubbers/vents? Typically when we map stuff in we use the version of scrubbers/vents that are off by default for the sake of lag reduction

Trilbyspaceclone commented on maps/__DeepTunnels/map/_Nadezhda_Deep_Tunnels.dmm:
```diff
@@ -8745,11 +8777,18 @@
 	},
 /area/nadezhda/rnd/mixing)
+"yc" = (
+/obj/machinery/atmospherics/unary/vent_scrubber/on{
```
We dont use active vents for sake of lag

ThePainkiller replied:
Fixed, thank you and CDB for pointing that one

Trilbyspaceclone commented on maps/__DeepTunnels/map/_Nadezhda_Deep_Tunnels.dmm:
```diff
@@ -9547,6 +9714,12 @@
 /turf/simulated/floor/tiled/steel,
 /area/mine/hallway)
+"VT" = (
+/obj/machinery/atmospherics/unary/vent_pump/on{
```
Same with these

Risikio commented:
Not exactly a fan of your vendomat fix, and I'm not actually sure it really works. I think it just gives the appearance of working, while the variables that matter aren't changed.

Did you VV and verify that the account goes to the correct account after it has been claimed?

ThePainkiller commented:
> Not exactly a fan of your vendomat fix, and I'm not actually sure it really works. I think it just gives the appearance of working, while the variables that matter aren't changed.
> Did you VV and verify that the account goes to the correct account after it has been claimed?

Yes. I wouldn't claim it's fixed otherwise.

![image](https://user-images.githubusercontent.com/1294508/175792410-14b7224a-e44c-49b7-b80c-21f80cbec5b4.png)

Trilbyspaceclone commented on code/game/machinery/kitchen/smartfridge.dm:
```diff
@@ -137,6 +137,7 @@
 /obj/machinery/smartfridge/drinks
 	name = "\improper drink showcase"
 	desc = "A refrigerated storage unit for tasty tasty alcohol."
+	density = 0 // For placing atop tables as a proper storage, like on the old layout. - Seb
```
```suggestion
	density = FALSE // For placing atop tables as a proper storage, like on the old layout. - Seb
```

Trilbyspaceclone commented on code/game/machinery/vendor/food_vendors.dm:
```diff
@@ -450,13 +450,25 @@
 	vend_delay = 15
 	idle_power_usage = 211 //refrigerator - believe it or not, this is actually the average power consumption of a refrigerated vending machine according to NRCan.
-	vendor_department = DEPARTMENT_LSS
+	density = 0 // Removing the duplicately defined vendor_department var is a bandaid fix for it being previously inaccesible by anyone. - Seb
```
```suggestion
	density = FALSE // Removing the duplicately defined vendor_department var is a bandaid fix for it being previously inaccesible by anyone. - Seb
```

Trilbyspaceclone commented on maps/__Nadezhda/map/_Nadezhda_Colony_Omni_STU.dmm:
```diff
@@ -143,13 +143,13 @@
 "aaK" = (
-/obj/machinery/camera/network/research{
-	dir = 8
+/obj/machinery/atmospherics/unary/vent_pump/on{
```
These are still on by default

Trilbyspaceclone commented:
drafted do to changes being on a local branch not yet pushed

ThePainkiller commented:
## Changes since last update: 

### Applies Xenobio suggestion by olive (#Noid1000)

<details> <summary> Expand </summary>

![NEW_Xenobio](https://user-images.githubusercontent.com/1294508/176129678-aa542ce8-c3ed-4b7b-be15-3dd124e8ff8f.png)

> Slimes can break reinforced windows if their charge reaches a critical stage, hence windows get replaced with reinforced plasma ones, windoors replaced by reinforced ones, all with Xenobio/flora access edited on them.

</details>

### Addresses player's concerns on machine placement in RND

<details> <summary> Expand </summary>

![NEW_RND_Robo](https://user-images.githubusercontent.com/1294508/176130524-6e897cf8-008e-4cfc-98f9-c8ae9a7b4f5c.png)

> RND now has the machines alligned near the main desk, moving the entrance of Robotics one tile to the right, rearranging the area it took. Now machines rest close to the client, without the scientist having to go back and forth to print the requested goods. This also alters the wiring of RND, Robotics and the Testing Range to be better connecting and avoid crisscrossing wires by the Xenoflora entrance with maintenance feed, making repairs easier and less headachy for the Guild.

</details>

### NEW: Xenoflora maint exit

<details> <summary> Expand </summary>

![NEW_xenofloramaint](https://user-images.githubusercontent.com/1294508/176131200-5013b776-c004-405b-8dfa-454b37d77c3e.png)

> Since robotics lost its maint entrance, adding one to Xenoflora for ease of exiting/trauma team entering from maintenance for quicker patient arrival should the case arise. Also repositions cameras and lights to avoid blind spots and give better illumination.

</details>

### Atmos and lightning improvements on research hallway/chemistry area

<details> <summary> Expand </summary>

![NEW_atmosimprov](https://user-images.githubusercontent.com/1294508/176131582-20e763e6-615a-4a5e-8f5a-257441351afd.png)

> Lights are now part of research hallway, thus they start on without obscuring it or having to have chemistry manned to compensate. Better spaced air pumps and scrubbers for the areas closed within firelocks for easier scrubbing should a plasmaflood happen.

</details>

### Circuits area touchup

<details> <summary> Expand </summary>

![NEW_Circuits](https://user-images.githubusercontent.com/1294508/176132022-fe3f9e4f-a960-4f79-9ecf-5bb516ac7e47.png)


![image](https://user-images.githubusercontent.com/1294508/176135043-086b424e-3f2f-4197-9ae4-2b38f892d80d.png)

> Adds the missing Build-a-Drone vendor, previously missing from the map remake. Deletes a camera mistakenly linked to Engineering network and replaces it with a colony underground one on the north. Lightbulb next to it for better lightning. Fixes camera blind spots on the hallway leading to plasmatag, rewires the hallway to avoid crisscrossing of wires to avoid mapping issues and making repairs more cohesive. Adds proper markings for extinguisher and MiniPharma vendor.

</details>

### Surgery OSHA touchup

<details> <summary> Expand </summary>

![NEW_Surgery](https://user-images.githubusercontent.com/1294508/176132845-f6d2296d-f361-440d-9826-04a6c0114086.png)

> Surgery gets its missing fire alarm, and a proper marking for its fire extinguisher on its south side.

</details>

### Library improvements!

<details> <summary> Expand </summary>

![NEW_Library](https://user-images.githubusercontent.com/1294508/176132978-3869757e-0f08-44d3-bb13-ca97a9960da6.png)

> Replaces the curtains with shutters for privacy, considering tinted windows work weird (pressing a button on one side tints both sides' windows...) and people could open the curtains from the outside. Moves around the light, APC and fire extinguisher for cohesion and better lightning, since bookcases block light. Adds an area blurb and short area ambience music tracks to the inside of the Library for better ambientation.

</details>

ThePainkiller commented:
## Final fixes and tweaks

### Robotic storage is its own area now

<details> <summary> Expand </summary>

![FINAL_robostorage](https://user-images.githubusercontent.com/1294508/176349201-e6b2a052-6e36-4111-a643-c5949e869f08.png)

> New area declared for robotic storage, should lessen power consumption of Robotics APC by being its own area. Added propper air alarm and APC as well as fire measures already present. Removes a missplaced camera on the lower right corner, as well as a missplaced air vent on the firing range.

</details>

### Research Lab renamed to Research Hallway

<details> <summary> Expand </summary>

![FINAL_hallwayapc](https://user-images.githubusercontent.com/1294508/176351918-6318a6af-b00d-44fc-9b49-a5e095c6a4b6.png)

> As present on the previous screenshot, the APC got replaced by an air alarm for the area. The APC is located now on the small storage room to the east, with proper flooring. The area was renamed as it was the only instance across every map where this area in particular was used, and more than a Lab, it was a Hallway separating Medical from Science. This also solves the badly positioned APC on the storage area, and changes its map icon to research proper.

</details>

### Final Xenobio touchup

<details> <summary> Expand </summary>

![FINAL_Xenobio](https://user-images.githubusercontent.com/1294508/176352205-2fd007b8-9fb1-4ebf-bcf8-a48c4a026f38.png)

> Xenobio's doors got replaced with actual Science airlocks, both solid and glass, as external airlocks have a bugged opacity when opened once and are in a weird state of being translucent-yet-not. Moves the Xenobio APC further inside the laboratory instead of the middle of the hallway, and adds additional lightbulbs for better lightning on the surgery area and xenoflora hallway.

</details>
